session.go

```diff
@@ -607,15 +608,17 @@ func createSessionWithDomainFunc(store kv.Storage) func(*domain.Domain) (pools.R
 
 func drainRecordSet(ctx context.Context, rs ast.RecordSet) ([]types.Row, error) {
 	var rows []types.Row
-	for {
-		row, err := rs.Next(ctx)
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
-		if row == nil {
-			break
-		}
-		rows = append(rows, row)
+	chk := rs.NewChunk()
+	err := rs.NextChunk(ctx, chk)
+	if err != nil {
+		return rows, errors.Trace(err)
+	}
+	if chk.NumRows() == 0 {
+		return rows, nil
+	}
+	iter := chunk.NewIterator4Chunk(chk)
+	for r := iter.Begin(); r != iter.End(); r = iter.Next() {
+		rows = append(rows, r)
 	}
 	return rows, nil
```

Comment:
I saw CI failed. Should I remove this line 623  return rows, nil ?

Reply:
@hsqlu From CI:
```txt
go get github.com/kisielk/errcheck
go get github.com/golang/lint/golint
vet
session.go:625: unreachable code
Makefile:75: recipe for target 'check' failed
make: *** [check] Error 1
Exited with code 2
```

actually you should remove line 625

Reply:
Yes, the line number above is unmatched and I fixed it in appending commit.